### PR TITLE
fix: support generating random IDs outside of secure contexts

### PR DESCRIPTION
**Pull request by hunterloftis**
## Description

Replace HTTPS-only `crypto.randomUUID` with `crypto.getRandomValues`, generating equivalent output (an 8-character hex string).

## Motivation and context

Allows SWC to run without errors in HTTP contexts.

## How has this been tested?

- All existing tests pass
- Added a test to validate the shape of the ID
- Added a test to validate the speed of ID generation

## Types of changes

-   [X] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [X] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [X] My code follows the code style of this project.
-   [X] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [X] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [X] I have added tests to cover my changes.
-   [X] All new and existing tests passed.
-   [X] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

**Comment by Westbrook** [comment]
This looks good, generally. With the breaking changes in Coach Mark, it's difficult to get full performance testing in the PR. If you're interested in fast forwarding this, feel free to run and paste the results from `yarn test:bench -p action-menu picker overlay split-button` to help clarify the changes herein. We've got a release coming up on Monday or Tuesday, so you could also wait till that is in and we can rebase this to a testable place.

**Comment by hunterloftis** [comment]
```
[----------------------------------------------------------] launching chromeRunning benchmarks

[==========================================================] done

┌─────────────┬────────────────────────┐
│   Benchmark │ action-menu:test-basic │
├─────────────┼────────────────────────┤
│     Browser │ chrome-headless        │
│             │ 121.0.6167.160         │
├─────────────┼────────────────────────┤
│ Sample size │ 50                     │
└─────────────┴────────────────────────┘

┌─────────┬────────────┬───────────────────┬───────────────────┬───────────────────┐
│ Version │ Bytes      │          Avg time │         vs remote │               vs  │
├─────────┼────────────┼───────────────────┼───────────────────┼───────────────────┤
│ remote  │ 640.44 KiB │ 72.93ms - 73.65ms │                   │            unsure │
│         │            │                   │          -        │         -2% - +0% │
│         │            │                   │                   │ -1.48ms - +0.13ms │
├─────────┼────────────┼───────────────────┼───────────────────┼───────────────────┤
│ <none>  │ 631.40 KiB │ 73.25ms - 74.69ms │            unsure │                   │
│         │            │                   │         -0% - +2% │          -        │
│         │            │                   │ -0.13ms - +1.48ms │                   │
└─────────┴────────────┴───────────────────┴───────────────────┴───────────────────┘

[----------------------------------------------------------] launching chromeRunning benchmarks

[==========================================================] done

┌─────────────┬───────────────────┐
│   Benchmark │ picker:basic-test │
├─────────────┼───────────────────┤
│     Browser │ chrome-headless   │
│             │ 121.0.6167.160    │
├─────────────┼───────────────────┤
│ Sample size │ 50                │
└─────────────┴───────────────────┘

┌─────────┬────────────┬─────────────────────┬─────────────────┬─────────────────┐
│ Version │ Bytes      │            Avg time │       vs remote │             vs  │
├─────────┼────────────┼─────────────────────┼─────────────────┼─────────────────┤
│ remote  │ 504.52 KiB │ 244.92ms - 247.02ms │                 │          faster │
│         │            │                     │        -        │         0% - 1% │
│         │            │                     │                 │ 0.47ms - 3.23ms │
├─────────┼────────────┼─────────────────────┼─────────────────┼─────────────────┤
│ <none>  │ 496.85 KiB │ 246.93ms - 248.71ms │          slower │                 │
│         │            │                     │         0% - 1% │        -        │
│         │            │                     │ 0.47ms - 3.23ms │                 │
└─────────┴────────────┴─────────────────────┴─────────────────┴─────────────────┘

[==========================================================] done

┌─────────────┬────────────────────┐
│   Benchmark │ overlay:basic-test │
├─────────────┼────────────────────┤
│     Browser │ chrome-headless    │
│             │ 121.0.6167.160     │
├─────────────┼────────────────────┤
│ Sample size │ 50                 │
└─────────────┴────────────────────┘

┌─────────┬────────────┬───────────────────┬───────────────────┬───────────────────┐
│ Version │ Bytes      │          Avg time │         vs remote │               vs  │
├─────────┼────────────┼───────────────────┼───────────────────┼───────────────────┤
│ remote  │ 506.05 KiB │ 30.58ms - 30.91ms │                   │            unsure │
│         │            │                   │          -        │         -1% - +1% │
│         │            │                   │                   │ -0.22ms - +0.29ms │
├─────────┼────────────┼───────────────────┼───────────────────┼───────────────────┤
│ <none>  │ 493.96 KiB │ 30.52ms - 30.90ms │            unsure │                   │
│         │            │                   │         -1% - +1% │          -        │
│         │            │                   │ -0.29ms - +0.22ms │                   │
└─────────┴────────────┴───────────────────┴───────────────────┴───────────────────┘

[----------------------------------------------------------] launching chromeRunning benchmarks

[==========================================================] done

┌─────────────┬─────────────────────────┐
│   Benchmark │ split-button:basic-test │
├─────────────┼─────────────────────────┤
│     Browser │ chrome-headless         │
│             │ 121.0.6167.160          │
├─────────────┼─────────────────────────┤
│ Sample size │ 50                      │
└─────────────┴─────────────────────────┘

┌─────────┬────────────┬───────────────────────┬───────────────────┬───────────────────┐
│ Version │ Bytes      │              Avg time │         vs remote │               vs  │
├─────────┼────────────┼───────────────────────┼───────────────────┼───────────────────┤
│ remote  │ 705.95 KiB │ 1714.93ms - 1725.26ms │                   │            unsure │
│         │            │                       │          -        │         -0% - +0% │
│         │            │                       │                   │ -8.14ms - +7.85ms │
├─────────┼────────────┼───────────────────────┼───────────────────┼───────────────────┤
│ <none>  │ 696.02 KiB │ 1714.14ms - 1726.35ms │            unsure │                   │
│         │            │                       │         -0% - +0% │          -        │
│         │            │                       │ -7.85ms - +8.14ms │                   │
└─────────┴────────────┴───────────────────────┴───────────────────┴───────────────────┘

action-menu:test-basic: 72.93ms - 73.65ms
action-menu:test-basic: 73.25ms - 74.69ms
picker:basic-test: 244.92ms - 247.02ms
picker:basic-test: 246.93ms - 248.71ms
overlay:basic-test: 30.58ms - 30.91ms
overlay:basic-test: 30.52ms - 30.90ms
split-button:basic-test: 1714.93ms - 1725.26ms
split-button:basic-test: 1714.14ms - 1726.35ms
✨  Done in 292.34s.
```

File: tools/shared/src/random-id.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+// This gnarly-looking implementation returns the equivalent of crypto.randomUUID().slice(0, 8).
+// It uses getRandomValues() in order to be compatible with HTTP contexts.
+export function randomID(): string {
```
Comment (Westbrook) [inline tools/shared/src/random-id.ts:15]: Thanks for weaving this into a helper method, I'd never know how to write this myself!

**Review by Westbrook — APPROVED** [review]
LGTM! :shipit: 

Can you please whip up an issue to leverage this pattern, instead of instance counting, in `AlertDialog.ts`, `FieldLabel.ts`, `HelpTextManager.ts`, `Tab.ts` and `TabPanel.ts`.

File: tools/shared/src/random-id.ts (hunk 1 above)
Comment (hunterloftis) [reply tools/shared/src/random-id.ts:15]: This is one of the cases where the tests _actually_ came first for me, since this took a few tries to get right!

**Comment by hunterloftis** [comment]
@Westbrook tracking here: https://github.com/adobe/spectrum-web-components/issues/4031

File: tools/shared/src/random-id.ts (hunk 1 above)
Comment (hunterloftis) [reply tools/shared/src/random-id.ts:15]: Also once this is in & part of the performance tests, I'd like to make this more readable by breaking out its components into named assignments in series. That _may_ have no impact on its speed, but we won't know without measuring.

**Pull request merged**